diff --git a/core/src/main/java/org/geysermc/geyser/entity/type/Entity.java b/core/src/main/java/org/geysermc/geyser/entity/type/Entity.java
index dd198930f..8b33c543e 100644
--- a/core/src/main/java/org/geysermc/geyser/entity/type/Entity.java
+++ b/core/src/main/java/org/geysermc/geyser/entity/type/Entity.java
@@ -322,9 +322,12 @@ public class Entity implements GeyserEntity {
                 moveEntityPacket.setHeadYaw(headYaw);
                 moveEntityPacket.getFlags().add(MoveEntityDeltaPacket.Flag.HAS_HEAD_YAW);
             }
-            if (isOnGround) {
+            boolean hasMovementOrRotation = !moveEntityPacket.getFlags().isEmpty();
+            if (isOnGround && hasMovementOrRotation) {
                 moveEntityPacket.getFlags().add(MoveEntityDeltaPacket.Flag.ON_GROUND);
             }
-            session.sendUpstreamPacket(moveEntityPacket);
+            if (hasMovementOrRotation) {
+                session.sendUpstreamPacket(moveEntityPacket);
+            }
         }
     }
